server/datastore/mysql/apple_mdm.go

```diff
@@ -1046,3 +1046,95 @@ func (ds *Datastore) UpdateHostMDMAppleProfile(ctx context.Context, profile *fle
         `, profile.Status, profile.OperationType, profile.Detail, profile.HostUUID, profile.CommandUUID)
 	return err
 }
+
+func (ds *Datastore) GetMDMAppleHostsProfilesSummary(ctx context.Context, teamID *uint) (*fleet.MDMAppleHostsProfilesSummary, error) {
+	// TODO(sarah): add cases to handle Fleet-managed profiles (e.g., disk encryption)
+	sqlFmt := `
+SELECT
+	count(us.host_uuid) AS count,
+	us.status
+FROM (
+	SELECT
+		host_uuid,
+		status
+	FROM
+		host_mdm_apple_profiles
+	WHERE
+		status = 'failed'
+	UNION
+	SELECT
+		host_uuid,
+		status
+	FROM
+		host_mdm_apple_profiles hmap
+	WHERE
+		status = 'pending'
+		AND host_uuid NOT IN(
+			SELECT
+				hmap2.host_uuid FROM host_mdm_apple_profiles hmap2
+			WHERE
+				hmap.host_uuid = hmap2.host_uuid
+				AND hmap2.status = 'failed')
+	UNION
+	SELECT
+		host_uuid, 
+		status
+	FROM
+		host_mdm_apple_profiles hmap
+	WHERE
+		status = 'applied'
+		AND host_uuid NOT IN(
+			SELECT
+				hmap2.host_uuid FROM host_mdm_apple_profiles hmap2
+			WHERE
+				hmap.host_uuid = hmap2.host_uuid
+				AND(hmap2.status = 'pending'
+					OR hmap2.status = 'failed'))) us
+WHERE
+	EXISTS (
+		SELECT
+			1
+		FROM
+			hosts h
+		WHERE
+			h.uuid = us.host_uuid
+			AND %s)
+	GROUP BY
+		status
+	`
```

Comment:
I wonder what's the `EXPLAIN` of this, will depend on the optimizer but seems like we're doing various filters that can be done at once? The other day I mentioned using `CASE`, I had in mind this:

```sql
SELECT
    COUNT(CASE WHEN h.failed > 0 THEN 'failed' END) as failed,
    COUNT(CASE WHEN h.failed = 0 AND h.pending > 0 THEN 'pending' END) as pending,
    COUNT(CASE WHEN h.failed =0 AND h.pending = 0 THEN 'applied' END) as applied       
FROM (
	SELECT
	  host_uuid,
	  COUNT(CASE WHEN status = 'applied' THEN 1 END) as applied,
	  COUNT(CASE WHEN status = 'failed' THEN 1 END) as failed,
	  COUNT(CASE WHEN status = 'pending' THEN 1 END) as pending
	FROM host_mdm_apple_profiles hmap 
	GROUP BY host_uuid
) as h
```

Reply:
cc: @mna do you have any ideas/thoughts? Sorry for the multiple pings, I'm trying to squish as much knowledge as we can since it's Wednesday and you'll probably not be around by the time we review this PR!

Reply:
Oh that looks great, should be faster based on intuition, but sometimes sql dbs can be quite counter-intuitive regarding performance (to me at least). Probably worth loading something like 1000 hosts locally (e.g. with `osquery-perf`) and testing the query time and explain plan.

Reply:
w00t! let's go with that one.

I'm confused why `hosts.uuid` will make any difference in the query with the `CASE` statements? it's only `SELECT`in from `host_mdm_apple_profiles` 🤔

Reply:
We also need a team filter on the count, which requires a sub select from `hosts`

https://github.com/fleetdm/fleet/pull/10194/files#diff-2f8a6d69f21403d6845da98a6a9a7e3808f275a21ec0209cbabcdbeaae57676eR1084

Reply:
Ah, it is because I put that in the wrong place. It should be after the GROUP BY. That brings the two much closer to parity without the index.

Reply:
ah, understood! we really need to add that index

Reply:
> Ah, it is because I put that in the wrong place. It should be after the GROUP BY. That brings the two much closer to parity without the index.

good catch! yeah, that should speed things a bit (still need to add that index tho 😭 )

Reply:
hmm, thinking more about it... I think it's good in its current place, it might benefit in your current simulation, but it's good that we're filtering before doing the aggregation right?

I'm spitballing just for fun here, so please feel free to ignore, but I wonder if we could `JOIN` with `mdm_apple_configuration_profiles` instead? it has the `team_id` we're looking for and should be way smaller than the number of hosts for big deployments where performance matters.

Reply:
I think that should work. I had considered it before and now I can't remember why I didn't go in that direction 🙃

Reply:
lol, yeah. We can iterate later if we find any perf issues, let's merge this as it looks great!